### PR TITLE
RDBC-8 Change Exception when throw DatabaseDoesNotExistsException in …

### PR DESCRIPTION
**Pull request by IdanHaim**
…FirstTopologyUpdate

### CODE REVIEW

File: src/Raven.Client/Http/RequestExecutor.cs [hunk 1]
```diff
@@ -478,8 +484,13 @@ private void UpdateTopologyCallback(object _)
 
             _lastKnownUrls = initialUrls;
 
-            throw new AggregateException("Failed to retrieve cluster topology from all known nodes" + Environment.NewLine +
-                                         string.Join(Environment.NewLine, list.Select(x => x.Item1 + " -> " + x.Item2?.Message))
+            ThrowExceptions(list);
+        }
+
+        protected virtual void ThrowExceptions(List<(string, Exception)> list)
+        {
```
Comment (ppekrol) [inline src/Raven.Client/Http/RequestExecutor.cs:491]: Leaving my comment from previous PR: https://github.com/ravendb/ravendb/pull/4356#discussion_r152026309 for further discussion.

File: src/Raven.Client/Http/RequestExecutor.cs [hunk 2]
```diff
@@ -442,6 +442,12 @@ private void UpdateTopologyCallback(object _)
                     _lastKnownUrls = initialUrls;
                     throw;
                 }
+                catch (DatabaseDoesNotExistException e)
+                {
+                    if (initialUrls.Length == 1)
```
Comment (ayende) [inline src/Raven.Client/Http/RequestExecutor.cs:447]: This is wrong, this should always throw. 
`DatabaseDoesNotExistsException` is not recoverable.

**Review by ayende — CHANGES_REQUESTED** [review]

File: src/Raven.Client/Http/RequestExecutor.cs (hunk 2 above)
Comment (IdanHaim) [reply src/Raven.Client/Http/RequestExecutor.cs:447]: What happens if you have more then one url in the store, and in one of those url the database exists?

Comment (IdanHaim) [reply src/Raven.Client/Http/RequestExecutor.cs]: What happens if you have more then one url in the store, and in one of those url the database exists?

**Pull request merged**